File: Company Profile/TugasAkhir/AnakKost/app/src/main/java/com/anak/anakkost/MainActivity.java
package com.anak.anakkost;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent(getApplicationContext(), RegistrasiVerifikasiActivity.class);
                startActivity(intent);
                finish();
            }
        }, 2000);

        findViewById(R.id.id).setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View view) {
                Intent intent = new Intent(getApplicationContext(), AdminActivity.class);
                startActivity(intent);
                finish();
                return false;
            }
        });
    }
}

File: Jet Pack Compose/Jakal7/Jakal7/app/src/main/java/com/willi16/jakal7/MainActivity.kt
package com.willi16.jakal7

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import com.google.firebase.auth.FirebaseAuth
import com.willi16.jakal7.databinding.ActivityMainBinding
import com.willi16.jakal7.home.HomeScreenActivity
import com.willi16.jakal7.login.SignInActivity

class MainActivity : AppCompatActivity() {
    private lateinit var mainBinding: ActivityMainBinding
    private lateinit var auth: FirebaseAuth
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mainBinding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(mainBinding.root)

        auth = FirebaseAuth.getInstance()
        val user = auth.currentUser

        Handler(Looper.myLooper()!!).postDelayed({
            if (user != null){
                val homeIntent = Intent(this, HomeScreenActivity::class.java)
                startActivity(homeIntent)
                finishAffinity()
            }else{
                val signIntent = Intent(this, SignInActivity::class.java)
                startActivity(signIntent)
                finish()
            }
        }, 3000)
    }
}